### PR TITLE
BAEL-5036: Create a Jandex index in Quarkus for classes in a external module

### PR DESCRIPTION
**Pull request by ueberfuhr**
Add Quarkus Hello App that uses several external modules that contain CDI managed beans. Each module is indexed using a different kind of registration.

### CODE REVIEW

**Comment by KevinGilmore** [comment]
retest this please

File: quarkus-jandex/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Relevant Articles:
```
Comment (KevinGilmore) [inline quarkus-jandex/README.md:1]: We can omit this README file for now. The editor will create it after the article is published to ensure that it has the correct URL.

**Comment by ueberfuhr** [comment]
retest this please

**Comment by ueberfuhr** [comment]
retest this please

**Pull request merged**